src/zui/ZUICreatePerson/PersonalInfoForm.tsx

```diff
@@ -0,0 +1,278 @@
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { ExpandMore } from '@mui/icons-material';
+import utc from 'dayjs/plugin/utc';
+import {
+  Box,
+  Button,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+} from '@mui/material';
+import dayjs, { Dayjs } from 'dayjs';
+import { FC, useEffect, useRef } from 'react';
+
+import globalMessageIds from 'core/i18n/globalMessageIds';
+import { makeNaiveDateString } from 'utils/dateUtils';
+import messageIds from 'zui/l10n/messageIds';
+import { TagManagerSection } from 'features/tags/components/TagManager';
+import useCustomFields from 'features/profile/hooks/useCustomFields';
+import { useNumericRouteParams } from 'core/hooks';
+import useTags from 'features/tags/hooks/useTags';
+import { checkInvalidFields, ShowAllTriggeredType } from '.';
+import { Msg, useMessages } from 'core/i18n';
+import { ZetkinCreatePerson, ZetkinTag } from 'utils/types/zetkin';
+
+dayjs.extend(utc);
+
+type GenderKeyType = 'f' | 'm' | 'o' | 'unknown';
+type ZetkinCreatePersonFields = keyof Omit<
+  ZetkinCreatePerson,
+  'tags' | 'customFields'
+>;
+
+interface PersonalInfoFormProps {
+  debounced: (
+    field: keyof ZetkinCreatePerson,
+    value: string | null,
+    custom: boolean
+  ) => void;
+  personalInfo: ZetkinCreatePerson;
+  onClickShowAll: (value: ShowAllTriggeredType) => void;
+  showAllClickedType: ShowAllTriggeredType;
```

Comment:
Whose responsibility is it to decide whether to show all the fields or not? As I see it here, it seems to be the responsibility both of the inner and the outer components, because the concept of showing all is expressed both in a callback (from child to parent) and in a regular property (from parent to child). That to me seems like a lack of separation of concerns, probably making the code harder to work with.